### PR TITLE
Javalab Neighborhood prefactor

### PR DESCRIPTION
**Pull request by jmkulwik**
This sets us up with a visualization column under the instructions area so we can add in Neighborhood (i.e. Maze).

With this, we get horizontal resize-ability.

We also needed to add in a custom mode for the instructions panel that allows the panel to work separately from the code workspace. In the past, the visualization resize area would look for any element that has the class "editor-column" put it on the right side of the screen, and adjust its "left" CSS property. In this case, we wanted the instructions panel on the opposite side of the screen and to be resized along with the rest of the visualization column.

ResizeVisualization

## Links

- [Engineering Plan](https://docs.google.com/document/d/1SqSLogXsOKJD8QexbZAbxCpUGRdNP9Yj7zW8tN6oADI/edit#)
- jira ticket: [CSA-285](https://codedotorg.atlassian.net/browse/CSA-285)
- [Spec](https://docs.google.com/document/d/1zVCVa2MRvMQqt1m13_KEP8vjWcf61Rd8N_u3wuuic04/edit#bookmark=id.innbr258uj3j)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/javalab/JavalabView.jsx [hunk 1]
```diff
@@ -120,20 +133,26 @@ class JavalabView extends React.Component {
...
-            <InstructionsWithWorkspace>
-              <div style={style.preview}>
-                <PaneHeader hasFocus={true}>
-                  <PaneSection>Preview</PaneSection>
-                </PaneHeader>
-              </div>
-            </InstructionsWithWorkspace>
+          <div
+            id="visualizationColumn"
+            className="with_padding responsive pin_bottom"
```
Comment (maureensturgeon) [inline apps/src/javalab/JavalabView.jsx]: Most classnames I've come across are kebab case. Do we have a convention for casing of class names?

File: apps/src/templates/instructions/TopInstructions.jsx [hunk 2]
```diff
@@ -670,7 +672,7 @@ class TopInstructions extends Component {
     return (
       <div
         style={topInstructionsStyle}
-        className="editor-column"
+        className={!standalone && 'editor-column'}
```
Comment (maureensturgeon) [inline apps/src/templates/instructions/TopInstructions.jsx]: I don't feel strongly about changing this, but another option here would be to use the classnames library `!standalone && 'editor-column'` = `classnames({'editor-column': !standalone})`

Comment (maureensturgeon) [reply apps/src/templates/instructions/TopInstructions.jsx]: A unit test for this change would be helpful!

**Review by maureensturgeon — APPROVED** [review]
The screenshot in the description is very helpful, thanks! I just left a couple notes, nothing blocking.

**Review by molly-moen — APPROVED** [review]
LGTM!

File: apps/src/javalab/JavalabView.jsx (hunk 1 above)
Comment (jmkulwik) [reply apps/src/javalab/JavalabView.jsx]: Looks like those two classes aren't needed anyway. Removed!

File: apps/src/templates/instructions/TopInstructions.jsx (hunk 2 above)
Comment (jmkulwik) [reply apps/src/templates/instructions/TopInstructions.jsx]: 👍

**Pull request merged**